consensus/propagation/commitment_state.go

```diff
@@ -173,25 +184,17 @@ func (p *ProposalCache) DeleteRound(height int64, round int32) {
 	}
 }
 
-// prune keeps the past X proposals / blocks in memory while deleting the rest.
-func (p *ProposalCache) prune(keepRecentHeights, keepRecentRounds int) {
+// prune deletes all cached compact blocks for heights less than the provided
+// height and round.
+//
+// todo: also prune rounds. this requires prune in the consensus reactor after
+// moving rounds.
+func (p *ProposalCache) prune(pruneHeight int64) {
 	p.pmtx.Lock()
 	defer p.pmtx.Unlock()
 	for height := range p.proposals {
-		if height < p.currentHeight-int64(keepRecentHeights) {
+		if height < pruneHeight {
 			delete(p.proposals, height)
 		}
 	}
-	// delete all but the last round for each remaining height except the current.
-	// this is because we need to keep the last round for the current height.
-	for height := range p.proposals {
-		if height == p.currentHeight {
-			continue
-		}
-		for round := range p.proposals[height] {
-			if round <= p.currentRound-int32(keepRecentRounds) {
-				delete(p.proposals[height], round)
-			}
-		}
-	}
```

Comment:
simplified pruning via just pruning everything after the consensus reactor commits to a block (instead of pruning after we receive new compact blocks. This makes pruning rounds more difficult tho